<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Create Regional Association Plots • locusplotr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Create Regional Association Plots">
<meta property="og:description" content="Description: This package allows the user to obtain linkage disequlibrium data for any contiguous region within the genome using the LocusZoom API \href{https://portaldev.sph.umich.edu/}{https://portaldev.sph.umich.edu/}. By integrating this LD information with results from a genome-wide association study (GWAS) the user can create regional association plots.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">locusplotr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mglev1n/locusplotr/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="locusplotr-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#locusplotr-" class="anchor"></a>locusplotr <img src="reference/figures/logo.png" align="right" height="139">
</h1></div>
<!-- badges: start -->

<p>The goal of <code>locusplotr</code> is to allow users to integrate genome-wide association study results with linkage disequilibrium data to create regional association plots surrounding a genomic locus of interest. The <code>gg_locusplot</code> function allows the user to provide GWAS summary statistics for a region of interest and a reference/sentinel/lead variant within that region, and will return a ggplot object with a regional association plot. In the background, this package uses a helper function named <code>ld_extract_locuszoom</code> to query the University of Michigan LocusZoom API (<a href="https://portaldev.sph.umich.edu/" class="uri">https://portaldev.sph.umich.edu/</a>) to obtain linkage disequilibrium data for a genetic variant and genomic region of interest (allowing the user to specify genome build and genetic ancestry).</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install <code>locusplotr</code> from <a href="https://github.com/">GitHub</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"mglev1n/locusplotr"</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>Start with a dataframe containing genome wide association study summary statistics at a locus of interest</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mglev1n.github.io/locusplotr">locusplotr</a></span><span class="op">)</span>

<span class="va">fto_locus_df</span>
<span class="co">#&gt; # A tibble: 19,119 × 9</span>
<span class="co">#&gt;    chromosome position rsid    effect_allele other_allele    eaf  effect std_err</span>
<span class="co">#&gt;         &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1         16 53532568 rs1503… A             C            0.996  -0.279   0.196 </span>
<span class="co">#&gt;  2         16 54016452 rs1722… A             G            0.909  -0.0269  0.0122</span>
<span class="co">#&gt;  3         16 53710931 rs1454… A             G            0.982  -0.0212  0.104 </span>
<span class="co">#&gt;  4         16 54069480 rs3717… T             C            0.001  -1.38    1.04  </span>
<span class="co">#&gt;  5         16 53371872 rs5337… T             C            0.0019  0.0681  0.508 </span>
<span class="co">#&gt;  6         16 53838423 rs1162… T             C            0.0221  0.0116  0.0985</span>
<span class="co">#&gt;  7         16 53610642 rs1131… T             C            0.149   0.0364  0.0099</span>
<span class="co">#&gt;  8         16 53494581 rs5350… A             G            0.997   0.264   0.170 </span>
<span class="co">#&gt;  9         16 54248221 rs5576… A             T            0.0476 -0.0527  0.0709</span>
<span class="co">#&gt; 10         16 53549981 &lt;NA&gt;    A             G            0.0004  1.50    0.518 </span>
<span class="co">#&gt; # … with 19,109 more rows, and 1 more variable: p_value &lt;dbl&gt;</span></code></pre></div>
<p>Use the <code>gg_locusplot</code> function to create a regional association plot. This calls the <code>ld_extract_locuszoom</code> function in the background to fetch linkage disequilibrium data from the University of Michigan LocusZoom API (<a href="https://portaldev.sph.umich.edu/" class="uri">https://portaldev.sph.umich.edu/</a>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span>

<span class="va">fto_locus_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/gg_locusplot.html">gg_locusplot</a></span><span class="op">(</span>
    lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>,
    rsid <span class="op">=</span> <span class="va">rsid</span>,
    chrom <span class="op">=</span> <span class="va">chromosome</span>,
    pos <span class="op">=</span> <span class="va">position</span>,
    ref <span class="op">=</span> <span class="va">effect_allele</span>,
    alt <span class="op">=</span> <span class="va">other_allele</span>,
    p_value <span class="op">=</span> <span class="va">p_value</span>
  <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-2-1.png" width="80%" style="display: block; margin: auto;"></p>
<p>Use the <code>plot_genes</code> argument to include a plot of genes within the region beneath the main regional association plot.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/gg_locusplot.html">gg_locusplot</a></span><span class="op">(</span>
  df <span class="op">=</span> <span class="va">fto_locus_df</span>,
  lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>,
  rsid <span class="op">=</span> <span class="va">rsid</span>,
  chrom <span class="op">=</span> <span class="va">chromosome</span>,
  pos <span class="op">=</span> <span class="va">position</span>,
  ref <span class="op">=</span> <span class="va">effect_allele</span>,
  alt <span class="op">=</span> <span class="va">other_allele</span>,
  p_value <span class="op">=</span> <span class="va">p_value</span>,
  plot_genes <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-3-1.png" width="80%" style="display: block; margin: auto;"> Additional arguments allow the user to specify the genome build (GRCh37, GRCH38), population (Eg. ALL, AMR, AFR, EUR, EAS, etc.), plot title/subtitle, and directory to save the regional association plot as a .pdf.<br><br><br></p>
<p>The <code>ld_extract_locuszoom</code> function, called in the background by <code>gg_locusplot</code>, can also be used to retrieve linkage disequilibrium data for a genetic variant and genomic region of interest. The user may again specify genome build, population, and the specific linkage disequilibrium metric (Eg. r, rsquare, cov)</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/ld_extract_locuszoom.html">ld_extract_locuszoom</a></span><span class="op">(</span>chrom <span class="op">=</span> <span class="fl">16</span>, pos <span class="op">=</span> <span class="fl">53830055</span>, ref <span class="op">=</span> <span class="st">"C"</span>, alt <span class="op">=</span> <span class="st">"G"</span>, start <span class="op">=</span> <span class="fl">53830055</span> <span class="op">-</span> <span class="fl">5e5</span>, stop <span class="op">=</span> <span class="fl">53830055</span> <span class="op">+</span> <span class="fl">5e5</span>, genome_build <span class="op">=</span> <span class="st">"GRCh37"</span>, population <span class="op">=</span> <span class="st">"ALL"</span>, metric <span class="op">=</span> <span class="st">"rsquare"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 28,455 × 7</span>
<span class="co">#&gt;    variant1        chromosome1 position1 variant2 chromosome2 position2 correlation</span>
<span class="co">#&gt;    &lt;chr&gt;                 &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;          &lt;dbl&gt;     &lt;int&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 16:53830055_C/G          16  53830055 16:5333…          16  53330083   0.000613 </span>
<span class="co">#&gt;  2 16:53830055_C/G          16  53830055 16:5333…          16  53330091   0.00328  </span>
<span class="co">#&gt;  3 16:53830055_C/G          16  53830055 16:5333…          16  53330128   0.0000874</span>
<span class="co">#&gt;  4 16:53830055_C/G          16  53830055 16:5333…          16  53330136   0.0000592</span>
<span class="co">#&gt;  5 16:53830055_C/G          16  53830055 16:5333…          16  53330148   0.00364  </span>
<span class="co">#&gt;  6 16:53830055_C/G          16  53830055 16:5333…          16  53330184   0.000674 </span>
<span class="co">#&gt;  7 16:53830055_C/G          16  53830055 16:5333…          16  53330263   0.000118 </span>
<span class="co">#&gt;  8 16:53830055_C/G          16  53830055 16:5333…          16  53330336   0.0000592</span>
<span class="co">#&gt;  9 16:53830055_C/G          16  53830055 16:5333…          16  53330401   0.000618 </span>
<span class="co">#&gt; 10 16:53830055_C/G          16  53830055 16:5333…          16  53330445   0.00202  </span>
<span class="co">#&gt; # … with 28,445 more rows</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/mglev1n/locusplotr/">https://​github.com/​mglev1n/​locusplotr/​</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Michael Levin <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9937-9932" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/mglev1n/locusplotr/actions"><img src="https://github.com/mglev1n/locusplotr/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Michael Levin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
