<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Create Regional Association Plots • locusplotr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Create Regional Association Plots">
<meta name="description" content="Description: This package allows the user to obtain linkage disequlibrium data for any contiguous region within the genome using the LocusZoom API \href{https://portaldev.sph.umich.edu/}{https://portaldev.sph.umich.edu/}. By integrating this LD information with results from a genome-wide association study (GWAS) the user can create regional association plots.">
<meta property="og:description" content="Description: This package allows the user to obtain linkage disequlibrium data for any contiguous region within the genome using the LocusZoom API \href{https://portaldev.sph.umich.edu/}{https://portaldev.sph.umich.edu/}. By integrating this LD information with results from a genome-wide association study (GWAS) the user can create regional association plots.">
<meta property="og:image" content="https://mglev1n.github.io/locusplotr/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">locusplotr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mglev1n/locusplotr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="locusplotr-">locusplotr <a class="anchor" aria-label="anchor" href="#locusplotr-"></a>
</h1>
</div>
<!-- badges: start -->

<p>The goal of <code>locusplotr</code> is to allow users to integrate genome-wide association study results with linkage disequilibrium data to create regional association plots surrounding a genomic locus of interest. The <code>gg_locusplot</code> function allows the user to provide GWAS summary statistics for a region of interest and a reference/sentinel/lead variant within that region, and will return a ggplot object with a regional association plot. In the background, this package uses a helper function named <code>ld_extract_locuszoom</code> to query the University of Michigan LocusZoom API (<a href="https://portaldev.sph.umich.edu/" class="external-link uri">https://portaldev.sph.umich.edu/</a>) to obtain linkage disequilibrium data for a genetic variant and genomic region of interest (allowing the user to specify genome build and genetic ancestry).</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <code>locusplotr</code> from <a href="https://github.com/" class="external-link">GitHub</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"mglev1n/locusplotr"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Start with a dataframe containing genome wide association study summary statistics at a locus of interest</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mglev1n.github.io/locusplotr">locusplotr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">fto_locus_df</span></span>
<span><span class="co">#&gt; # A tibble: 19,119 × 9</span></span>
<span><span class="co">#&gt;    chromosome position rsid    effect_allele other_allele    eaf  effect std_err</span></span>
<span><span class="co">#&gt;         &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1         16 53532568 rs1503… A             C            0.996  -0.279   0.196 </span></span>
<span><span class="co">#&gt;  2         16 54016452 rs1722… A             G            0.909  -0.0269  0.0122</span></span>
<span><span class="co">#&gt;  3         16 53710931 rs1454… A             G            0.982  -0.0212  0.104 </span></span>
<span><span class="co">#&gt;  4         16 54069480 rs3717… T             C            0.001  -1.38    1.04  </span></span>
<span><span class="co">#&gt;  5         16 53371872 rs5337… T             C            0.0019  0.0681  0.508 </span></span>
<span><span class="co">#&gt;  6         16 53838423 rs1162… T             C            0.0221  0.0116  0.0985</span></span>
<span><span class="co">#&gt;  7         16 53610642 rs1131… T             C            0.149   0.0364  0.0099</span></span>
<span><span class="co">#&gt;  8         16 53494581 rs5350… A             G            0.997   0.264   0.170 </span></span>
<span><span class="co">#&gt;  9         16 54248221 rs5576… A             T            0.0476 -0.0527  0.0709</span></span>
<span><span class="co">#&gt; 10         16 53549981 &lt;NA&gt;    A             G            0.0004  1.50    0.518 </span></span>
<span><span class="co">#&gt; # ℹ 19,109 more rows</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: p_value &lt;dbl&gt;</span></span></code></pre></div>
<p>Use the <code>gg_locusplot</code> function to create a regional association plot. This calls the <code>ld_extract_locuszoom</code> function in the background to fetch linkage disequilibrium data from the University of Michigan LocusZoom API (<a href="https://portaldev.sph.umich.edu/" class="external-link uri">https://portaldev.sph.umich.edu/</a>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">fto_locus_df</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/gg_locusplot.html">gg_locusplot</a></span><span class="op">(</span></span>
<span>    lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>,</span>
<span>    rsid <span class="op">=</span> <span class="va">rsid</span>,</span>
<span>    chrom <span class="op">=</span> <span class="va">chromosome</span>,</span>
<span>    pos <span class="op">=</span> <span class="va">position</span>,</span>
<span>    ref <span class="op">=</span> <span class="va">effect_allele</span>,</span>
<span>    alt <span class="op">=</span> <span class="va">other_allele</span>,</span>
<span>    p_value <span class="op">=</span> <span class="va">p_value</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-2-1.png" width="80%" style="display: block; margin: auto;"></p>
<p>Use the <code>plot_genes</code> argument to include a plot of genes within the region beneath the main regional association plot.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/gg_locusplot.html">gg_locusplot</a></span><span class="op">(</span></span>
<span>  df <span class="op">=</span> <span class="va">fto_locus_df</span>,</span>
<span>  lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>,</span>
<span>  rsid <span class="op">=</span> <span class="va">rsid</span>,</span>
<span>  chrom <span class="op">=</span> <span class="va">chromosome</span>,</span>
<span>  pos <span class="op">=</span> <span class="va">position</span>,</span>
<span>  ref <span class="op">=</span> <span class="va">effect_allele</span>,</span>
<span>  alt <span class="op">=</span> <span class="va">other_allele</span>,</span>
<span>  p_value <span class="op">=</span> <span class="va">p_value</span>,</span>
<span>  plot_genes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-3-1.png" width="80%" style="display: block; margin: auto;"> Additional arguments allow the user to specify the genome build (GRCh37, GRCH38), population (Eg. ALL, AMR, AFR, EUR, EAS, etc.), plot title/subtitle, and directory to save the regional association plot as a .pdf.<br><br><br></p>
<p>The <code>ld_extract_locuszoom</code> function, called in the background by <code>gg_locusplot</code>, can also be used to retrieve linkage disequilibrium data for a genetic variant and genomic region of interest. The user may again specify genome build, population, and the specific linkage disequilibrium metric (Eg. r, rsquare, cov)</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/ld_extract_locuszoom.html">ld_extract_locuszoom</a></span><span class="op">(</span>chrom <span class="op">=</span> <span class="fl">16</span>, pos <span class="op">=</span> <span class="fl">53830055</span>, ref <span class="op">=</span> <span class="st">"C"</span>, alt <span class="op">=</span> <span class="st">"G"</span>, start <span class="op">=</span> <span class="fl">53830055</span> <span class="op">-</span> <span class="fl">5e5</span>, stop <span class="op">=</span> <span class="fl">53830055</span> <span class="op">+</span> <span class="fl">5e5</span>, genome_build <span class="op">=</span> <span class="st">"GRCh37"</span>, population <span class="op">=</span> <span class="st">"ALL"</span>, metric <span class="op">=</span> <span class="st">"rsquare"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 28,455 × 7</span></span>
<span><span class="co">#&gt;    variant1     chromosome1 position1 variant2 chromosome2 position2 correlation</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;              &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;          &lt;dbl&gt;     &lt;int&gt;       &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 16:53830055…          16  53830055 16:5333…          16  53330083   0.000613 </span></span>
<span><span class="co">#&gt;  2 16:53830055…          16  53830055 16:5333…          16  53330091   0.00328  </span></span>
<span><span class="co">#&gt;  3 16:53830055…          16  53830055 16:5333…          16  53330128   0.0000874</span></span>
<span><span class="co">#&gt;  4 16:53830055…          16  53830055 16:5333…          16  53330136   0.0000592</span></span>
<span><span class="co">#&gt;  5 16:53830055…          16  53830055 16:5333…          16  53330148   0.00364  </span></span>
<span><span class="co">#&gt;  6 16:53830055…          16  53830055 16:5333…          16  53330184   0.000674 </span></span>
<span><span class="co">#&gt;  7 16:53830055…          16  53830055 16:5333…          16  53330263   0.000118 </span></span>
<span><span class="co">#&gt;  8 16:53830055…          16  53830055 16:5333…          16  53330336   0.0000592</span></span>
<span><span class="co">#&gt;  9 16:53830055…          16  53830055 16:5333…          16  53330401   0.000618 </span></span>
<span><span class="co">#&gt; 10 16:53830055…          16  53830055 16:5333…          16  53330445   0.00202  </span></span>
<span><span class="co">#&gt; # ℹ 28,445 more rows</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/mglev1n/locusplotr/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing locusplotr</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Michael Levin <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9937-9932" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/mglev1n/locusplotr/actions" class="external-link"><img src="https://github.com/mglev1n/locusplotr/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://github.com/mglev1n/locusplotr/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/mglev1n/locusplotr/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Michael Levin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
