<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create a regional association plot — gg_locusplot • locusplotr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Create a regional association plot — gg_locusplot"><meta property="og:description" content="Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API https://portaldev.sph.umich.edu/) for that locus to create a regional association plot."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">locusplotr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mglev1n/locusplotr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a regional association plot</h1>
    <small class="dont-index">Source: <a href="https://github.com/mglev1n/locusplotr/blob/HEAD/R/gg_locusplot.R" class="external-link"><code>R/gg_locusplot.R</code></a></small>
    <div class="hidden name"><code>gg_locusplot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API <a href="https://portaldev.sph.umich.edu/" class="external-link">https://portaldev.sph.umich.edu/</a>) for that locus to create a regional association plot.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gg_locusplot</span><span class="op">(</span></span>
<span>  <span class="va">df</span>,</span>
<span>  lead_snp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rsid <span class="op">=</span> <span class="va">rsid</span>,</span>
<span>  chrom <span class="op">=</span> <span class="va">chrom</span>,</span>
<span>  pos <span class="op">=</span> <span class="va">pos</span>,</span>
<span>  ref <span class="op">=</span> <span class="va">ref</span>,</span>
<span>  alt <span class="op">=</span> <span class="va">alt</span>,</span>
<span>  p_value <span class="op">=</span> <span class="va">p_value</span>,</span>
<span>  trait <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_pvalue_threshold <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  plot_subsample_prop <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  plot_distance <span class="op">=</span> <span class="fl">5e+05</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="st">"GRCh37"</span>,</span>
<span>  population <span class="op">=</span> <span class="st">"ALL"</span>,</span>
<span>  plot_genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plot_title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_subtitle <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>df</dt>
<dd><p>Dataframe containing columns with rsid, chromosome, position, reference/effect allele, alternate/non-effect allele, and p-value for all variants within the range of interest</p></dd>


<dt>lead_snp</dt>
<dd><p>A character vector containing a lead variant of interest. When NULL (default), the variant with the lowest p-value will be selected as the lead variant.</p></dd>


<dt>rsid</dt>
<dd><p>Rsid column</p></dd>


<dt>chrom</dt>
<dd><p>Chromosome column</p></dd>


<dt>pos</dt>
<dd><p>Position column</p></dd>


<dt>ref</dt>
<dd><p>Reference/effect allele column</p></dd>


<dt>alt</dt>
<dd><p>Alternate/non-effect allele column</p></dd>


<dt>p_value</dt>
<dd><p>P-value column</p></dd>


<dt>trait</dt>
<dd><p>(optional) Column containing the name of the trait</p></dd>


<dt>plot_pvalue_threshold</dt>
<dd><p>Threshold for plotting p-value on regional association plot (default = 0.1) - reducing the number of points decreases file size and improves performance</p></dd>


<dt>plot_subsample_prop</dt>
<dd><p>Proportion of points above p-value threshold to plot (default = 0.1; range = 0-1) - reducing the number of points decreases file size and improves performance</p></dd>


<dt>plot_distance</dt>
<dd><p>Integer corresponding to the size of the locus that should be plotted</p></dd>


<dt>genome_build</dt>
<dd><p>Character - one of "GRCh37" or "GRCh38"</p></dd>


<dt>population</dt>
<dd><p>Character - one of "ALL", "AFR", "AMR", "EAS", "EUR", "SAS" referring to the reference population of interest for obtaining linkage disequilibrium information (default = "ALL")</p></dd>


<dt>plot_genes</dt>
<dd><p>Logical - Include a plot of genes/transcripts within the region of interest beneath the regional association plot (default = FALSE)</p></dd>


<dt>plot_title</dt>
<dd><p>A character string corresponding to plot title (default = NULL)</p></dd>


<dt>plot_subtitle</dt>
<dd><p>A character string corresponding to plot subtitle (default = NULL)</p></dd>


<dt>path</dt>
<dd><p>Character string (default = NULL) - if a path is supplied a .pdf of the plot will be saved</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A ggplot object containing a regional association plot for the locus of interest</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># Basic regional association plot</span></span></span>
<span class="r-in"><span><span class="fu">gg_locusplot</span><span class="op">(</span>df <span class="op">=</span> <span class="va">fto_locus_df</span>, lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>, rsid <span class="op">=</span> <span class="va">rsid</span>, chrom <span class="op">=</span> <span class="va">chromosome</span>, pos <span class="op">=</span> <span class="va">position</span>, ref <span class="op">=</span> <span class="va">effect_allele</span>, alt <span class="op">=</span> <span class="va">other_allele</span>, p_value <span class="op">=</span> <span class="va">p_value</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use "plot_genes = TRUE" to add a plot of genes within the region beneath the regional association plot</span></span></span>
<span class="r-in"><span><span class="fu">gg_locusplot</span><span class="op">(</span>df <span class="op">=</span> <span class="va">fto_locus_df</span>, lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>, rsid <span class="op">=</span> <span class="va">rsid</span>, chrom <span class="op">=</span> <span class="va">chromosome</span>, pos <span class="op">=</span> <span class="va">position</span>, ref <span class="op">=</span> <span class="va">effect_allele</span>, alt <span class="op">=</span> <span class="va">other_allele</span>, p_value <span class="op">=</span> <span class="va">p_value</span>, plot_genes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Michael Levin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

