<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Create a regional association plot — gg_locusplot • locusplotr</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create a regional association plot — gg_locusplot"><meta name="description" content="Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API https://portaldev.sph.umich.edu/) for that locus to create a regional association plot."><meta property="og:description" content="Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API https://portaldev.sph.umich.edu/) for that locus to create a regional association plot."><meta property="og:image" content="https://mglev1n.github.io/locusplotr/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">locusplotr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.5.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mglev1n/locusplotr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Create a regional association plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/mglev1n/locusplotr/blob/main/R/gg_locusplot.R" class="external-link"><code>R/gg_locusplot.R</code></a></small>
      <div class="d-none name"><code>gg_locusplot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API <a href="https://portaldev.sph.umich.edu/" class="external-link">https://portaldev.sph.umich.edu/</a>) for that locus to create a regional association plot.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gg_locusplot</span><span class="op">(</span></span>
<span>  <span class="va">df</span>,</span>
<span>  lead_snp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rsid <span class="op">=</span> <span class="va">rsid</span>,</span>
<span>  chrom <span class="op">=</span> <span class="va">chrom</span>,</span>
<span>  pos <span class="op">=</span> <span class="va">pos</span>,</span>
<span>  ref <span class="op">=</span> <span class="va">ref</span>,</span>
<span>  alt <span class="op">=</span> <span class="va">alt</span>,</span>
<span>  effect <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  std_err <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  p_value <span class="op">=</span> <span class="va">p_value</span>,</span>
<span>  trait <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_pvalue_threshold <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  plot_subsample_prop <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  plot_distance <span class="op">=</span> <span class="fl">5e+05</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="st">"GRCh37"</span>,</span>
<span>  population <span class="op">=</span> <span class="st">"ALL"</span>,</span>
<span>  plot_genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plot_recombination <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plot_title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_subtitle <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-df">df<a class="anchor" aria-label="anchor" href="#arg-df"></a></dt>
<dd><p>Dataframe containing columns with rsid, chromosome, position, reference/effect allele, alternate/non-effect allele, and p-value for all variants within the range of interest</p></dd>


<dt id="arg-lead-snp">lead_snp<a class="anchor" aria-label="anchor" href="#arg-lead-snp"></a></dt>
<dd><p>A character vector containing a lead variant of interest. When NULL (default), the variant with the lowest p-value will be selected as the lead variant.</p></dd>


<dt id="arg-rsid">rsid<a class="anchor" aria-label="anchor" href="#arg-rsid"></a></dt>
<dd><p>Rsid column</p></dd>


<dt id="arg-chrom">chrom<a class="anchor" aria-label="anchor" href="#arg-chrom"></a></dt>
<dd><p>Chromosome column</p></dd>


<dt id="arg-pos">pos<a class="anchor" aria-label="anchor" href="#arg-pos"></a></dt>
<dd><p>Position column</p></dd>


<dt id="arg-ref">ref<a class="anchor" aria-label="anchor" href="#arg-ref"></a></dt>
<dd><p>Reference/effect allele column</p></dd>


<dt id="arg-alt">alt<a class="anchor" aria-label="anchor" href="#arg-alt"></a></dt>
<dd><p>Alternate/non-effect allele column</p></dd>


<dt id="arg-effect">effect<a class="anchor" aria-label="anchor" href="#arg-effect"></a></dt>
<dd><p>Effect size column (on beta or log-odds scale)</p></dd>


<dt id="arg-std-err">std_err<a class="anchor" aria-label="anchor" href="#arg-std-err"></a></dt>
<dd><p>Standard error column</p></dd>


<dt id="arg-p-value">p_value<a class="anchor" aria-label="anchor" href="#arg-p-value"></a></dt>
<dd><p>P-value column</p></dd>


<dt id="arg-trait">trait<a class="anchor" aria-label="anchor" href="#arg-trait"></a></dt>
<dd><p>(optional) Column containing the name of the trait</p></dd>


<dt id="arg-plot-pvalue-threshold">plot_pvalue_threshold<a class="anchor" aria-label="anchor" href="#arg-plot-pvalue-threshold"></a></dt>
<dd><p>Threshold for plotting p-value on regional association plot (default = 0.1) - reducing the number of points decreases file size and improves performance</p></dd>


<dt id="arg-plot-subsample-prop">plot_subsample_prop<a class="anchor" aria-label="anchor" href="#arg-plot-subsample-prop"></a></dt>
<dd><p>Proportion of points above p-value threshold to plot (default = 0.25; range = 0-1) - reducing the number of points decreases file size and improves performance</p></dd>


<dt id="arg-plot-distance">plot_distance<a class="anchor" aria-label="anchor" href="#arg-plot-distance"></a></dt>
<dd><p>Integer corresponding to the size of the locus that should be plotted</p></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>Character - one of "GRCh37" or "GRCh38"</p></dd>


<dt id="arg-population">population<a class="anchor" aria-label="anchor" href="#arg-population"></a></dt>
<dd><p>Character - one of "ALL", "AFR", "AMR", "EAS", "EUR", "SAS" referring to the reference population of interest for obtaining linkage disequilibrium information (default = "ALL")</p></dd>


<dt id="arg-plot-genes">plot_genes<a class="anchor" aria-label="anchor" href="#arg-plot-genes"></a></dt>
<dd><p>Logical - Include a plot of genes/transcripts within the region of interest beneath the regional association plot (default = FALSE)</p></dd>


<dt id="arg-plot-recombination">plot_recombination<a class="anchor" aria-label="anchor" href="#arg-plot-recombination"></a></dt>
<dd><p>Logical - Include a secondary y-axis of recombination rate within the region of interest</p></dd>


<dt id="arg-plot-title">plot_title<a class="anchor" aria-label="anchor" href="#arg-plot-title"></a></dt>
<dd><p>A character string corresponding to plot title (default = NULL)</p></dd>


<dt id="arg-plot-subtitle">plot_subtitle<a class="anchor" aria-label="anchor" href="#arg-plot-subtitle"></a></dt>
<dd><p>A character string corresponding to plot subtitle (default = NULL)</p></dd>


<dt id="arg-path">path<a class="anchor" aria-label="anchor" href="#arg-path"></a></dt>
<dd><p>Character string (default = NULL) - if a path is supplied a .pdf of the plot will be saved</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A ggplot object containing a regional association plot for the locus of interest</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Basic regional association plot</span></span></span>
<span class="r-in"><span><span class="fu">gg_locusplot</span><span class="op">(</span>df <span class="op">=</span> <span class="va">fto_locus_df</span>, lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>, rsid <span class="op">=</span> <span class="va">rsid</span>, chrom <span class="op">=</span> <span class="va">chromosome</span>, pos <span class="op">=</span> <span class="va">position</span>, ref <span class="op">=</span> <span class="va">effect_allele</span>, alt <span class="op">=</span> <span class="va">other_allele</span>, p_value <span class="op">=</span> <span class="va">p_value</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use "plot_genes = TRUE" to add a plot of genes within the region beneath the regional association plot</span></span></span>
<span class="r-in"><span><span class="fu">gg_locusplot</span><span class="op">(</span>df <span class="op">=</span> <span class="va">fto_locus_df</span>, lead_snp <span class="op">=</span> <span class="st">"rs62033413"</span>, rsid <span class="op">=</span> <span class="va">rsid</span>, chrom <span class="op">=</span> <span class="va">chromosome</span>, pos <span class="op">=</span> <span class="va">position</span>, ref <span class="op">=</span> <span class="va">effect_allele</span>, alt <span class="op">=</span> <span class="va">other_allele</span>, p_value <span class="op">=</span> <span class="va">p_value</span>, plot_genes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Michael Levin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

