<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create a regional association plot — gg_locusplot • locusplotr</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Create a regional association plot — gg_locusplot" />
<meta property="og:description" content="Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API https://portaldev.sph.umich.edu/) for that locus to create a regional association plot." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">locusplotr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mglev1n/locusplotr/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a regional association plot</h1>
    <small class="dont-index">Source: <a href='https://github.com/mglev1n/locusplotr/blob/master/R/gg_locusplot.R'><code>R/gg_locusplot.R</code></a></small>
    <div class="hidden name"><code>gg_locusplot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Returns a ggplot object containing a regional association plot (-log10(p-value) as a function of chromosomal position, with variants colored by linkage disequilibrium to reference variant).
This function allows the user to integrate genome wide association study (GWAS) summary statistics for a locus of interest with linkage disequilibrium information (obtained using the University of Michigan LocusZoom API <a href='https://portaldev.sph.umich.edu/'>https://portaldev.sph.umich.edu/</a>) for that locus to create a regional association plot.</p>
    </div>

    <pre class="usage"><span class='fu'>gg_locusplot</span><span class='op'>(</span>
  <span class='va'>df</span>,
  lead_snp <span class='op'>=</span> <span class='cn'>NULL</span>,
  rsid <span class='op'>=</span> <span class='va'>rsid</span>,
  chrom <span class='op'>=</span> <span class='va'>chrom</span>,
  pos <span class='op'>=</span> <span class='va'>pos</span>,
  ref <span class='op'>=</span> <span class='va'>ref</span>,
  alt <span class='op'>=</span> <span class='va'>alt</span>,
  p_value <span class='op'>=</span> <span class='va'>p_value</span>,
  plot_pvalue_threshold <span class='op'>=</span> <span class='fl'>0.1</span>,
  plot_subsample_prop <span class='op'>=</span> <span class='fl'>0.1</span>,
  plot_distance <span class='op'>=</span> <span class='fl'>5e+05</span>,
  genome_build <span class='op'>=</span> <span class='st'>"GRCh37"</span>,
  population <span class='op'>=</span> <span class='st'>"ALL"</span>,
  plot_genes <span class='op'>=</span> <span class='cn'>FALSE</span>,
  plot_title <span class='op'>=</span> <span class='cn'>NULL</span>,
  plot_subtitle <span class='op'>=</span> <span class='cn'>NULL</span>,
  path <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>df</th>
      <td><p>Dataframe containing columns with rsid, chromosome, position, reference/effect allele, alternate/non-effect allele, and p-value for all variants within the range of interest</p></td>
    </tr>
    <tr>
      <th>lead_snp</th>
      <td><p>A character vector containing a lead variant of interest. When NULL (default), the variant with the lowest p-value will be selected as the lead variant.</p></td>
    </tr>
    <tr>
      <th>rsid</th>
      <td><p>Rsid column</p></td>
    </tr>
    <tr>
      <th>chrom</th>
      <td><p>Chromosome column</p></td>
    </tr>
    <tr>
      <th>pos</th>
      <td><p>Position column</p></td>
    </tr>
    <tr>
      <th>ref</th>
      <td><p>Reference/effect allele column</p></td>
    </tr>
    <tr>
      <th>alt</th>
      <td><p>Alternate/non-effect allele column</p></td>
    </tr>
    <tr>
      <th>p_value</th>
      <td><p>P-value column</p></td>
    </tr>
    <tr>
      <th>plot_pvalue_threshold</th>
      <td><p>Threshold for plotting p-value on regional association plot (default = 0.1) - reducing the number of points decreases file size and improves performance</p></td>
    </tr>
    <tr>
      <th>plot_subsample_prop</th>
      <td><p>Proportion of points above p-value threshold to plot (default = 0.1; range = 0-1) - reducing the number of points decreases file size and improves performance</p></td>
    </tr>
    <tr>
      <th>plot_distance</th>
      <td><p>Integer corresponding to the size of the locus that should be plotted</p></td>
    </tr>
    <tr>
      <th>genome_build</th>
      <td><p>Character - one of "GRCh37" or "GRCh38"</p></td>
    </tr>
    <tr>
      <th>population</th>
      <td><p>Character - one of "ALL", "AFR", "AMR", "EAS", "EUR", "SAS" referring to the reference population of interest for obtaining linkage disequilibrium information (default = "ALL")</p></td>
    </tr>
    <tr>
      <th>plot_genes</th>
      <td><p>Logical - Include a plot of genes/transcripts within the region of interest beneath the regional association plot (default = FALSE)</p></td>
    </tr>
    <tr>
      <th>plot_title</th>
      <td><p>A character string corresponding to plot title (default = NULL)</p></td>
    </tr>
    <tr>
      <th>plot_subtitle</th>
      <td><p>A character string corresponding to plot subtitle (default = NULL)</p></td>
    </tr>
    <tr>
      <th>path</th>
      <td><p>Character string (default = NULL) - if a path is supplied a .pdf of the plot will be saved</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A ggplot object containing a regional association plot for the locus of interest</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='co'># Basic regional association plot</span>
<span class='fu'>gg_locusplot</span><span class='op'>(</span>df <span class='op'>=</span> <span class='va'>fto_locus_df</span>, lead_snp <span class='op'>=</span> <span class='st'>"rs62033413"</span>, rsid <span class='op'>=</span> <span class='va'>rsid</span>, chrom <span class='op'>=</span> <span class='va'>chromosome</span>, pos <span class='op'>=</span> <span class='va'>position</span>, ref <span class='op'>=</span> <span class='va'>effect_allele</span>, alt <span class='op'>=</span> <span class='va'>other_allele</span>, p_value <span class='op'>=</span> <span class='va'>p_value</span><span class='op'>)</span>

<span class='co'># Use "plot_genes = TRUE" to add a plot of genes within the region beneath the regional association plot</span>
<span class='fu'>gg_locusplot</span><span class='op'>(</span>df <span class='op'>=</span> <span class='va'>fto_locus_df</span>, lead_snp <span class='op'>=</span> <span class='st'>"rs62033413"</span>, rsid <span class='op'>=</span> <span class='va'>rsid</span>, chrom <span class='op'>=</span> <span class='va'>chromosome</span>, pos <span class='op'>=</span> <span class='va'>position</span>, ref <span class='op'>=</span> <span class='va'>effect_allele</span>, alt <span class='op'>=</span> <span class='va'>other_allele</span>, p_value <span class='op'>=</span> <span class='va'>p_value</span>, plot_genes <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='op'>}</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Michael Levin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


